File: SCTE65/src/java/other/org/himalay/si/MMS_Record.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.si;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Sun Sep 14 22:48:36 EDT 2014")
public class MMS_Record extends BinStruct implements PublicBinMsg {

	// members variables
	// bf1
	public BitField_8 bf1;
	// bf2
	public BitField_8 bf2;
	// symbolRate
	public long symbolRate;

	public MMS_Record() // throws Exception
	{
		init();
	}

	private void init() {
		// Initialize bf1
		bf1 = new BitField_8();
		// Initialize bf2
		bf2 = new BitField_8();
		// Initialize symbolRate

	}

	public int readNoHeader(DataInputStream istream) throws IOException {

		return read(istream);
	}

	public int read(DataInputStream istream) throws IOException {
		preRead();
		int retVal = 0;

		// read bf1
		retVal += bf1.read(istream);
		// read bf2
		retVal += bf2.read(istream);
		// read symbolRate
		{
			symbolRate = (long) (BinPrimitive.readUI32(istream));
			retVal += 4;
		}

		postRead();
		return retVal;
	}

	public int write(DataOutputStream ostream) throws IOException {
		preWrite();
		int retVal = 0;

		// write bf1
		ostream.writeByte(bf1.getValue());
		retVal += 1;
		// write bf2
		ostream.writeByte(bf2.getValue());
		retVal += 1;
		// write symbolRate
		BinPrimitive.writeUI32(ostream, symbolRate);
		retVal += 4;
		postWrite();
		return retVal;
	}

	public int dump(DumpContext dc) throws IOException {
		dc.indent();
		dc.getPs().print("MMS_Record\n");
		dc.increaseIndent();
		int retVal = 0;
		// write bf1
		{
			dc.indent();
			dc.getPs().print("transmissionSystem: ");
			dc.getPs().println(
					BitField_8.toDisplayString(getTransmissionSystem(), 4));
			dc.indent();
			dc.getPs().print("innerCodignMode: ");
			dc.getPs().println(
					BitField_8.toDisplayString(getInnerCodignMode(), 4));
		}
		// write bf2
		{
			dc.indent();
			dc.getPs().print("split_bitstream_mode: ");
			dc.getPs().println(
					BitField_8.toDisplayString(getSplit_bitstream_mode(), 1));
			dc.indent();
			dc.getPs().print("modulation_format: ");
			dc.getPs().println(
					BitField_8.toDisplayString(getModulation_format(), 5));
		}
		// write symbolRate
		dc.indent();
		dc.getPs().println(
				"symbolRate=" + symbolRate + "(0x"
						+ Long.toHexString(symbolRate) + ")");
		dc.decreaseIndent();
		return retVal;
	}

	// Getter for bf1
	// public BitField_8 getBf1()
	// {
	// return bf1 ;
	// }

	// Setter for bf1
	// public void setBf1(BitField_8 val)
	// {
	// this.bf1= val;
	// }
	// Getter for bf2
	// public BitField_8 getBf2()
	// {
	// return bf2 ;
	// }

	// Setter for bf2
	// public void setBf2(BitField_8 val)
	// {
	// this.bf2= val;
	// }
	// Getter for symbolRate
	// public long getSymbolRate()
	// {
	// return symbolRate ;
	// }

	// Setter for symbolRate
	// public void setSymbolRate(long val)
	// {
	// this.symbolRate= val;
	// }

	public int getTransmissionSystem() {
		return (bf1.getValue() & 0x000000f0) >> 4;
	}

	public void setTransmissionSystem(int val) {
		bf1.setValue((bf1.getValue() & 0xffffff0f) | ((val << 4) & 0x000000f0));
	}

	public int getInnerCodignMode() {
		return (bf1.getValue() & 0x0000000f) >> 0;
	}

	public void setInnerCodignMode(int val) {
		bf1.setValue((bf1.getValue() & 0xfffffff0) | ((val << 0) & 0x0000000f));
	}

	public int getSplit_bitstream_mode() {
		return (bf2.getValue() & 0x00000080) >> 7;
	}

	public void setSplit_bitstream_mode(int val) {
		bf2.setValue((bf2.getValue() & 0xffffff7f) | ((val << 7) & 0x00000080));
	}

	public int getModulation_format() {
		return (bf2.getValue() & 0x0000001f) >> 0;
	}

	public void setModulation_format(int val) {
		bf2.setValue((bf2.getValue() & 0xffffffe0) | ((val << 0) & 0x0000001f));
	}

	public int getSize() throws IOException {
		DataOutputStream dos = new DataOutputStream(new NullStream());
		return this.write(dos);
	}

}

// End of code

File: libsi/src/bp/MMS_Record.cpp
//**//
//@Created(date = "Mon Feb 01 22:29:58 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "MMS_Record.h"

#ifdef org_himalay_si_MMS_Record__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace si {

using namespace ::org::himalay::msgs::runtime;

// Constructor
MMS_Record::MMS_Record() {
    references= 0;


    // bf1
    bf1 = 0;
    // bf2
    bf2 = 0;
    // symbolRate
    symbolRate= (ui32)0;
}
// Destructor
MMS_Record::~MMS_Record() {
    // bf1

    // bf2

    // symbolRate

}

int MMS_Record::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // bf1
    bf1= istream.readUI8();
    retVal += 1;
    // bf2
    bf2= istream.readUI8();
    retVal += 1;
    // symbolRate
    {
        symbolRate=istream.readUI32() ;
        retVal+=4;
    }
    return retVal;
}

int MMS_Record::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read bf1
    bf1= istream.readUI8();
    retVal += 1;
    // read bf2
    bf2= istream.readUI8();
    retVal += 1;
    // read symbolRate
    {
        symbolRate=istream.readUI32() ;
        retVal+=4;
    }

    return retVal;
}

int MMS_Record::write(DataOutputStream& ostream) { // throws IOException




    int retVal= 0;
    // write bf1
    ostream.writeUI8(bf1);
    retVal +=1;
    // write bf2
    ostream.writeUI8(bf2);
    retVal +=1;
    // write symbolRate
    ostream.writeUI32(symbolRate);
    retVal +=4;

    return retVal;
}

int MMS_Record::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("MMS_Record\n");
    dc.increaseIndent();
    int retVal= 0;
    // write bf1
    {
        dc.indent();
        dc.getPs().print("transmissionSystem: ");
        dc.getPs().printlnBin(getTransmissionSystem(),4);
        dc.indent();
        dc.getPs().print("innerCodignMode: ");
        dc.getPs().printlnBin(getInnerCodignMode(),4);
    }
    // write bf2
    {
        dc.indent();
        dc.getPs().print("split_bitstream_mode: ");
        dc.getPs().printlnBin(getSplit_bitstream_mode(),1);
        dc.indent();
        dc.getPs().print("modulation_format: ");
        dc.getPs().printlnBin(getModulation_format(),5);
    }
    // write symbolRate
    dc.indent();
    dc.getPs().print("symbolRate=");
    dc.getPs().println((long)symbolRate);
    dc.decreaseIndent();
    return retVal;
}


// Getter for bf1
//ui8 MMS_Record::getBf1()
//{
//return this->bf1;
//}

// Setter for bf1
//void MMS_Record::setBf1(ui8 val)
//{
//this->bf1= val;
//}

// Getter for bf2
//ui8 MMS_Record::getBf2()
//{
//return this->bf2;
//}

// Setter for bf2
//void MMS_Record::setBf2(ui8 val)
//{
//this->bf2= val;
//}

// Getter for symbolRate
//ui32 MMS_Record::getSymbolRate()
//{
//return this->symbolRate;
//}

// Setter for symbolRate
//void MMS_Record::setSymbolRate(ui32 val)
//{
//this->symbolRate= val;
//}

int MMS_Record::getTransmissionSystem( ) {
    return ( this->bf1 & 0x000000f0 ) >> 4 ;
}

void MMS_Record::setTransmissionSystem(int val) {
    bf1=(  (this->bf1 & 0xffffff0f) | ( (val << 4 ) & 0x000000f0));
}

int MMS_Record::getInnerCodignMode( ) {
    return ( this->bf1 & 0x0000000f ) >> 0 ;
}

void MMS_Record::setInnerCodignMode(int val) {
    bf1=(  (this->bf1 & 0xfffffff0) | ( (val << 0 ) & 0x0000000f));
}

int MMS_Record::getSplit_bitstream_mode( ) {
    return ( this->bf2 & 0x00000080 ) >> 7 ;
}

void MMS_Record::setSplit_bitstream_mode(int val) {
    bf2=(  (this->bf2 & 0xffffff7f) | ( (val << 7 ) & 0x00000080));
}

int MMS_Record::getModulation_format( ) {
    return ( this->bf2 & 0x0000001f ) >> 0 ;
}

void MMS_Record::setModulation_format(int val) {
    bf2=(  (this->bf2 & 0xffffffe0) | ( (val << 0 ) & 0x0000001f));
}

int MMS_Record::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_si_MMS_Record__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::si::MMS_Record* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::si::MMS_Record* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code